### ISSUE TITLE
Rails 4.1 - Issue updating STI column

### ISSUE THREAD

**Issue by jmirmina** [airblade/paper_trail#404]
I've run into this weird issue running Rails 4.1.4 and paper_trail 3.0.3. Ill try and simplify my code to explain what is happening.

```
class Page < ActiveRecord::Base
  class Draft < ActiveRecord::Base
    include PaperTrail
    has_paper_trail
  end
end

class MainPage < Page
  class Draft < Page::Draft; end
end

class SubPage < MainPage
  class Draft < MainPage::Draft; end
end
```

The page and page_drafts table both have the 'type' inheritance column. There are some instances where I need to change the type. Previously, I was able to do something like this:
```
my_page = MainPage::Draft.create
my_page.update type: 'SubPage::Draft'
```
The update call would then update the type and updated_at columns. When upgrading to Rails 4.1.4, the update call is only updating the updated_at field. I noticed that when removing paper_trail from the Page::Draft class, it works as expected. 

The weird thing is that it seems to work as expected if my_page is an instance of the Page::Draft class where has_paper_trail is called. If my_page is an instance of a subclass of Page::Draft, thats when I notice this issue.

I apologize if this is hard to follow. I've been trying to use the template at https://github.com/rails/rails/blob/master/guides/bug_report_templates/active_record_master.rb to create a failing test, but I am having issues getting PaperTrail included.

I originally thought this was an issue with Rails 4.1 until I discovered that removing PaperTrail fixed my issue. Reverting to Rails 4.0.8 and leaving in PaperTrail also fixes the issue. Any idea what could be going on?

**Comment by gdomingu** [comment]
Having the same issue with 3.0.6

**Comment by TheKidCoder** [comment]
I'm also having this issue with `3.0.1`.

Really basic STI though. 
```ruby
class Prompt < ActiveRecord::Base
  has_paper_trail
end

class TextField < Prompt
end

class RadioButton < Prompt
end
```

Simply trying to do `TextField.first.update_attributes type: "RadioButton"` does not change the type.

**Comment by batter** [comment]
Sorry for the delayed response here but can one of you try running your tests against PaperTrail version 4.0.0.beta2..  It seems like the issue is solved on PaperTrail 4.x.  Here is a template based off @rdubya's submission from this thread that works fine for me locally:

```ruby
gem 'activerecord', '~> 4.1.1'
gem 'sqlite3'
gem 'paper_trail', '4.0.0.beta2'

require 'active_record'
require 'paper_trail'
require 'minitest/autorun'
require 'logger'

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts do |t|
    t.string :type
    t.string :title
    t.string :author
  end

  create_table :versions do |t|
    t.string :item_type, :null => false
    t.integer :item_id, :null => false
    t.string :event, :null => false
    t.integer :whodunnit
    t.text :object
    t.text :object_changes
    t.datetime :created_at
  end
end

# Papertrail nested classes
class Post < ActiveRecord::Base
  has_paper_trail
end

class NewsPost < Post; end
class BlogPost < Post; end

class BugTest < Minitest::Test
  def test_changing_type_with_inner_class
    post = NewsPost.create!
    post.update(type: 'BlogPost')
    updated_post = Post.find(post.id)
    assert_instance_of BlogPost, updated_post
  end
end
```

Not 100% certain but I'm guessing that this may have been fixed by the move for the update version recording callbacks from `before_` to `after_` (#375/#374).  Unless someone can reproduce this consistently then I believe this bug can be closed.

**Comment by batter** [comment]
Looks like this was actually fixed as a byproduct of the fix for #428. I am going to backport this change to PaperTrail `3.x

**Comment by batter** [comment]
Looks like this was actually fixed as a byproduct of the fix for #428 (contained in the `master` / PaperTrail `4.x` branch). I am going to backport this fix to PaperTrail `3.x`, and cut a `3.0.7` release containing the fix too.

**Comment by TheKidCoder** [comment]
Just wanted to follow up and confirm 4.x has fixed this issue!
